break retry loop when unable to start runner.py for n times

snegurok opened the pull request:
This is a workaround for #170. After n (say 10) unsuccessful retries of spawning runner.py, break the loop and allow the updater code to work.

I tested it spoiling runner.py and fixing it in repo after runner.sh falls into the loop. It was able to recover.

porthole-ascend-cinnamon commented:
It makes me question why we even need the inner loop..

porthole-ascend-cinnamon commented:
Shall we just replace the inner loop with the following code?

```
  if [ -z "$PID" ]
  then
    AUTO_MH=1 $PYTHON runner.py $SCRIPT_ARGS & PID=$!
    sleep 1
  fi

  if [ "${SCRIPT_ARGS}" == "--help" ]
  then
    exit 0
  fi

  if ! kill -0 $PID
  then
    PID=""
    echo -e "\n${RED}Error starting - retry in 30 seconds! Ctrl+C to exit${RESET}"
    sleep 30
  else
    sleep 600
  fi
```

snegurok commented:
The simpler the better.

porthole-ascend-cinnamon commented:
@snegurok would be grateful if you could make the changes and test it

snegurok commented:
Sure, gladly.

Tested 2 scenarios:
1. Spoiled python, fixed in repo. Correctly pulled and fixed.
2. Normal run, update in repo. Current run was killed and update rolled out.

I prepare a new pull request with that.
